src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -105,6 +110,57 @@ class JCRNodeDecorator {
         return definition.isMandatory()
     }
 
+    /**
+     * Build node and "only" mandatory child nodes
+     * @param decoratedNode
+     * @param mandatoryChildNodes
+     */
+    @Nonnull
+     ProtoNode buildNode() {
+        final ProtoNodeBuilder protoNodeBuilder = ProtoNode.newBuilder()
+        protoNodeBuilder.setName(path)
+        protoNodeBuilder.addAllProperties(getProtoProperties())
+        requiredChildNodes?.each {
+            protoNodeBuilder.addMandatoryChildNode(it.buildNode())
+        }
+        return protoNodeBuilder.build()
+    }
+
+    /**
+     * Build a collection of Proto Properties from JCR node
```

Comment:
Javadoc needs updating/trimming. To me, the explanation of the method doesn't provide additional value.